### ISSUE TITLE
add ordered transitions should skip states without children

### ISSUE THREAD

**Issue by d-e-e-p** [tyarkoni/transitions#156]
``` python

# in example below, **machine.add_ordered_transitions()** fails 
# but alternative function that skip hierarchical states works.

from subprocess import call

from transitions import logger
from transitions.extensions import MachineFactory
from transitions.extensions.nesting import NestedState

GraphMachine = MachineFactory.get_predefined(graph=True, nested=True)


states = ['A', 'B', {'name': 'C', 'children': ['1', '2',
    {'name': '3', 'children': ['a', 'b', 'c']}]}, 'D', 'E', 'F']


NestedState.separator = '/'
machine = GraphMachine(None, states, initial='A',
    auto_transitions=False,
    ignore_invalid_triggers=True,
)

def local_add_ordered_transitions(machine=None, trigger='next_state'):
    """
    only add ordered transitions for states without children 
    """

    states = []
    for (name, value) in machine.states.items():
        if (not value.children):
            states.append(name)

    for i in range(1, len(states)):
        machine.add_transition(trigger, states[i - 1], states[i])

    machine.add_transition(trigger, states[-1], states[0])



#machine.add_ordered_transitions()  # Gives Warning: cluster named cluster_C/3 not found
local_add_ordered_transitions(machine=machine) # works

machine.next_state()
print(machine.state)
machine.graph.draw('temp.jpg', prog='dot')
call(["tycat",'temp.jpg'])

```

**Comment by aleneum** [comment]
Hi @d-e-e-p,

thank you for your feedback. This is a bug which I also discovered recently. It has been tackled in pull request #151 . The subgraphs are wrongly named during the graph generation process. Unfortunately, This is not the only issue with nested state machines and graph generation at the moment.
Since I cannot tell when we will merge #151, I backported some bugfixes to the current `master` and added your example as a test case to `test_nesting`. See the result below.

> add ordered transitions should skip states without children

In my oppinion states without children should not be ignored. Transitions should be created from a) parents to their first child, from b) siblings to their next sibling and c) from last siblings to their parent's next sibling in this particular order. As you see in the image below, `transitions` follows this pattern at the moment. What you may also see is that right now transitions from parents to their first child are not visible in the graph.

So long story short: The graph may be inaccurate in certain states. Things should be more accurate when #151 is merged. However, if the graph generation in `master` throws errors, let us know anyway and we will do our best to fix it.

Best regards,
Alex

![temp](https://cloud.githubusercontent.com/assets/205986/19781906/d81aac5a-9c8b-11e6-98b6-888a70b8e5e0.jpg)
